src/system/SystemPacketBuffer.h

```diff
@@ -142,6 +144,20 @@ class DLL_EXPORT PacketBuffer : private pbuf
 #endif // CHIP_SYSTEM_CONFIG_USE_LWIP
 
 public:
+    void DebugDump(const char * title) const
+    {
+#ifdef DEBUG
+        printf("DebugDump %s message DataLength=%d\n", title, DataLength());
+        uint8_t startOfBuf[DataLength()];
+        Read(startOfBuf, sizeof(startOfBuf));
```

Comment:
Why do we need to read into that on-stack buffer?  Why can't we just directly log it from out internal buffer?  Would be much better.

Reply:
addressed in latest